### PR TITLE
"verification of custom FSID"

### PR DESCRIPTION
**Pull request by preethisn**
Signed-off-by: Preethi Nataraj <pnataraj@localhost.localdomain>

# Description - Verification of Custom FSID 

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs

# Checklist:

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarin Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test

Logs are copied at: magna002 /home/pnataraj/Logs

### CODE REVIEW

File: ceph/ceph_admin/bootstrap.py [hunk 1]
```diff
@@ -92,3 +98,19 @@ def bootstrap(self: CephAdmProtocol, config: Dict) -> None:
         logger.error("Bootstrap error: %s", err.read().decode())
 
         self.distribute_cephadm_gen_pub_key()
+
+        # Verification of arguments
+
+        args = self.config.get("args", {})
+        logger.info(args)
```
Comment (psathyan) [inline ceph/ceph_admin/bootstrap.py:105]: Could we remove this statement to reduce logging noise?

File: ceph/ceph_admin/bootstrap.py [hunk 2]
```diff
@@ -92,3 +98,19 @@ def bootstrap(self: CephAdmProtocol, config: Dict) -> None:
...
+        for key, value in args.items():
+            func = self.get_method(key)
+            if not func:
+                continue
+            if not func(value):
+                raise AssertionError
+
+    def validate_fsid(self, fsid) -> bool:
```
Comment (psathyan) [inline ceph/ceph_admin/bootstrap.py:113]: Please provide documentation for method.

File: ceph/ceph_admin/bootstrap.py [hunk 3]
```diff
@@ -14,6 +14,12 @@
 class BootstrapMixin:
     """Add bootstrap support to the child class."""
 
+    def get_method(self, key):
```
Comment (psathyan) [inline ceph/ceph_admin/bootstrap.py:17]: Could we have some documentation about the method?

File: ceph/ceph_admin/bootstrap.py [hunk 4]
```diff
@@ -92,3 +98,19 @@ def bootstrap(self: CephAdmProtocol, config: Dict) -> None:
...
+        for key, value in args.items():
+            func = self.get_method(key)
+            if not func:
+                continue
+            if not func(value):
+                raise AssertionError
+
+    def validate_fsid(self, fsid) -> bool:
+        out, err = self.shell(args=["ceph", "fsid"])
+        logger.info("Installed with custom fsid %s", fsid)
```
Comment (psathyan) [inline ceph/ceph_admin/bootstrap.py:115]: Do we need to log this statement?

**Review by psathyan — COMMENTED** [review]
Thanks for adding the verification for `fsid`.

File: ceph/ceph_admin/bootstrap.py (hunk 4 above)
Comment (preethisn) [reply ceph/ceph_admin/bootstrap.py:115]: @psathyan, Its not mandatory but I feel its good to print when something gets validated as messaging makes easier for us to understand when we go through the logs. Thoughts please.

File: ceph/ceph_admin/bootstrap.py (hunk 3 above)
Comment (preethisn) [reply ceph/ceph_admin/bootstrap.py:58]: Added doc string

**Comment by preethisn** [comment]
New logs are copied to http://magna002.ceph.redhat.com/ceph-qe-logs/Preethi/

**Review by psathyan — APPROVED** [review]
Looks good to me

**Pull request merged**